code/__HELPERS/icon_smoothing.dm

```diff
@@ -111,8 +111,10 @@
 /proc/smooth_icon(atom/A)
 	if(qdeleted(A))
 		return
-	spawn(0) //don't remove this, otherwise smoothing breaks
-		if(A && (A.smooth & SMOOTH_TRUE) || (A.smooth & SMOOTH_MORE))
+	if(!A || !A.smooth)
+		return
+	spawn(0)
```

Comment:
Same performance, but if I use spawn() I can avoid it if the if checks are true.